src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const isHerbivore = animal instanceof Herbivore;
+    const isHidden = animal.hidden === true;
+
+    animal.health = isHerbivore
+      ? (isHidden ? animal.health : animal.health - 50) : animal.health;
```

Comment:
NEVER format code like this (with double ?:)! 
It is impossible to read

The next one is much simpler

```suggestion
    if (isHerbivore && !animal.hidden && animal.health > 0) {
      animal.health -= 50;
    }
```